src/parse/rules/java_rules.build_defs

```diff
@@ -113,6 +114,91 @@ def java_library(name, srcs=None, src_dir=None, resources=None, resources_root=N
+
+    Args:
+      name (str): Name of the rule
+      srcs (list): Java source files to compile for this library
+      src_dir (str): Directory containing Java source files to compile.
+      resources (list): Resources to include in the .jar file
+      resources_root (str): Root directory to treat resources relative to; ie. if we are in
+                            //project/main/resources and resources_root is project/main then
+                            the resources in the .jar will be in the subdirectory 'resources'.
+      deps (list): Dependencies of this rule.
+      exported_deps (list): Exported dependencies, ie. dependencies that other things depending on this
+                            rule will also receive when they're compiling. This is quite important for
+                            Java; any dependency that forms part of the public API for your classes
+                            should be an exported dependency.
+      visibility (list): Visibility declaration of this rule.
+      test_only (bool): If True, this rule can only be depended on by tests.
+      javac_flags (list): List of flags passed to javac.
+    """
+    return java_library(name=name,
+                        srcs=srcs,
+                        src_dir=src_dir,
+                        resources=resources,
+                        resources_root=resources_root,
+                        deps=deps,
+                        visibility=visibility,
+                        test_only=test_only,
+                        javac_flags=javac_flags,
+                        modular=True)
+
+
+def java_runtime_image(name, main_module, main_class, modules, out=None,
+                       deps=None, data=None, visibility=None, jlink_args=None):
+    """Assembles a set of modules into an executable java runtime image.
+
+    Args:
+      name (str): Name of the rule.
+      main_module (str): Main module to set in the manifest. Has to be included in 'modules'.
+      main_class (str): Main class to set in the manifest. Has to belong to 'main_module'.
+      modules (list): Modules to be included in the runtime image.
+      out (str): Name of the folder that contains the runtime image and the binary contained by it. Defaults to 'name'.
+      deps (list): Dependencies of this rule.
+      data (list): Runtime data files for this rule.
+      visibility (list): Visibility declaration of this rule.
+      jlink_args (str): Arguments to pass to the JVM in the run script.
+    """
+    if not CONFIG.JAVA_HOME:
+        raise Exception('Java home needs to be set to link java runtime images against jmods.')
+    if not CONFIG.JLINK_TOOL:
+        raise Exception('A jlink tool is required to build java runtime images.')
+    if not modules:
+        raise Exception('You cannot assemble a java runtime image without specifying any modules.')
+    deps = deps or []
+    out = out or name
+    jlink_args = jlink_args or '--strip-debug --compress=2'
```

Comment:
If the default is a string, just make that the default argument value (we only avoid that for mutable things like lists which can get a bit weird)